### PR TITLE
dumpnef performance

### PR DESCRIPTION
**Pull request by Hecate2**
I spent almost 1 minute in dumpnef for a huge contract reaching instruction address of 22000. So I profiled dumpnef and made some optimizations...

### CODE REVIEW

File: src/Neo.Compiler.CSharp/Optimizer/DumpNef.cs [hunk 1]
```diff
@@ -320,14 +321,14 @@ public static string GenerateDumpNef(NefFile nef, JToken? debugInfo)
...
                                 else
                                     src = docPathToContent[docPath][lineIndex - 1].Trim();
-                                dumpnef += $"# Code {Path.GetFileName(docPath)} line {startLine}: \"{src}\"\n";
+                                dumpnef.Append($"# Code {Path.GetFileName(docPath)} line {startLine}: \"{src}\"\n");
                             }
                     }
                 }
                 if (a < script.Length)
-                    dumpnef += $"{WriteInstruction(a, script.GetInstruction(a), script.GetInstructionAddressPadding(), nef.Tokens)}\n";
+                    dumpnef.Append($"{WriteInstruction(a, script.GetInstruction(a), addressPadding, nef.Tokens)}\n");
```
[the remark this answers is not in the document]
Comment (Hecate2) [reply src/Neo.Compiler.CSharp/Optimizer/DumpNef.cs:329]: OK. WILCO in 24 hours

**Review by shargon — APPROVED** [review]

**Comment by Jim8y** [comment]
what is the performance improvement

**Comment by Hecate2** [comment]
> what is the performance improvement

string addressPadding enumerates instructions only once instead of many times
StringBuilder is much faster than string addition when there are many strings

**Comment by Jim8y** [comment]
> hat is the performance improvement

I mean what is the perforamcne after this pr.

**Comment by Hecate2** [comment]
> > hat is the performance improvement
> 
> I mean what is the perforamcne after this pr. 

< 1 second

**Pull request merged**